Rewrite LoginFragment

NN171 opened the pull request:
Added the values of Inter font, sign in button. The login screen was rewrote, also added a logo app.

ImmortalIdiot commented:
Result:
![image](https://github.com/ImmortalIdiot/RUTLead-project-activity/assets/127626768/6611faec-e762-492b-92b1-be36e07b58fd)
Please, change the colors.

ImmortalIdiot commented:
I specifically added the "Dimensions.kt" file so that we wouldn't write values in dp in the files, but use variables from the Dimensions.kt. Replace all dp values with variables from the file. Add other dimensions if necessary.

NN171 commented:
Changed it, also changed the pallete of DarkColorScheme and LightColorScheme. Added new dimensions values, replaced the dp values to dimensions values. Added a custom cursor. Fixed the button shape and its border

NN171 commented:
> Result: ![image](https://private-user-images.githubusercontent.com/127626768/309519748-6611faec-e762-492b-92b1-be36e07b58fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODc5MDgsIm5iZiI6MTcwOTU4NzYwOCwicGF0aCI6Ii8xMjc2MjY3NjgvMzA5NTE5NzQ4LTY2MTFmYWVjLWU3NjItNDkyYi05MmIxLWJlMzZlMDdiNThmZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQyMTI2NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzM0ZjI0ODg1YWU0NmE0N2JkMDYzMDM5NDI0NGRjMTM2M2ZjNzQ5ZjBiMWU0ZTU3YTZkYmM4MGExZmQzNmJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XSbk3hinBqMigZGhTwEW2MsSos1wJuTLekuqWcK6-GE) Please, change the colors.

It was changed

NN171 commented:
> I specifically added the "Dimensions.kt" file so that we wouldn't write values in dp in the files, but use variables from the Dimensions.kt. Replace all dp values with variables from the file. Add other dimensions if necessary.

Replaced all dp values.

NN171 commented:
Dp values was replaced by the dimensions values

NN171 commented on app/src/main/res/drawable/design_ic_visibility.xml:
Was done

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+        modifier = Modifier
+            .fillMaxWidth()
+            .onFocusChanged { isFocused = !isFocused }
+            .border(BorderStroke(3.dp, Color.Transparent),
```
Method BorderStroke() was removed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/auth/Login.kt:
```diff
@@ -0,0 +1,86 @@
+                    .fillMaxWidth(0.85f)
+                    .fillMaxHeight(0.6f)
+                    .clip(RoundedCornerShape(15.dp))
+                    .background(LightBlue),
```
Replaced by dimensions values

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                    Spacer(modifier = Modifier.height(dimensions.verticalXXLarge))
+                    SignInButton()
+                    Spacer(modifier = Modifier.height(dimensions.verticalXLarge))
+                    Row(modifier = Modifier
```
All parameters should be written on new lines.

NN171 replied:
Has been done

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+        derivedStateOf { text.isEmpty() }
+    }
+    val dimensions = LocalDimensions.current
+    CompositionLocalProvider(LocalTextSelectionColors provides customCursorHandleColor) {
```
What does this line do?

NN171 replied:
This line permits to change the handle of cursor color
![image](https://github.com/ImmortalIdiot/RUTLead-project-activity/assets/61082311/e976563e-17f4-42d9-8caa-db3a4589aae9) ![image](https://github.com/ImmortalIdiot/RUTLead-project-activity/assets/61082311/53f5e012-dc25-46bc-a23a-3f4d12391502)

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/AccountMissing.kt:
```diff
@@ -0,0 +1,20 @@
+        color = if (isSystemInDarkTheme()) DarkWhite
+        else Color.White
+    )
+}
```
Missing an empty line in the end of the file

NN171 replied:
Was added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/BoxLabel.kt:
```diff
@@ -0,0 +1,20 @@
+        textAlign = TextAlign.Center,
+        style = mediumInter32
+    )
+}
```
Missed an empty line in the end of the file

NN171 replied:
Was added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,42 @@
+    Button(
+        onClick = { /*TODO*/ },
+        modifier = Modifier
+            .height(50.dp)
```
Recommendation: replace the fixed size (in dp) with a size that depends on the parent container

NN171 replied:
Has been changed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,42 @@
+        onClick = { /*TODO*/ },
+        modifier = Modifier
+            .height(50.dp)
+            .width(150.dp)
```
Recommendation: replace the fixed size (in dp) with a size that depends on the parent container

NN171 replied:
Has been changed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,40 @@
+            ),
+        colors = ButtonDefaults.buttonColors(
+            containerColor = LightBlue
+        )) {
```
Move ") {" to the next line

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+    val icon = if(passwordVisible)
+        painterResource(id = R.drawable.password_visibility_on)
+    else
+        painterResource(id = R.drawable.password_visibility_off)
```
Missed an empty line after this line

NN171 replied:
The empty line after variable declaration has been added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+            onValueChange = { password = it },
+            label = {
+                Text(
+                    type,
```
What is this variable? Specifiy the parameters explicitly.
Example:
My function looks like this:
```Kotlin
fun doWork(
    firstAction: String = "",
    timeForFirstAction: Time = Time.ZERO,
    secondAction: String = "",
    timeForSecondAction: Time = Time.ZERO
) {
    // function body
}
```
When using a function with multiple arguments, you should specify the parameters explicitly.
I want to use my fun doWork() and specify only one parameter firstAction. I should explicitly write it to improve readability:
```Kotlin
val plans = Actions (firstAction = "Coding")
```

NN171 replied:
The param has been added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+            label = {
+                Text(
+                    type,
+                    style = if (!isFocused || !isFieldEmpty) TextStyle(
```
Replace the TextStyle container with the usual variable boldInter12 or mediumInter12

NN171 replied:
Has been changed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+                        fontSize = 12.sp,
+                        fontFamily = InterFontFamily
+                    )
+                    else TextStyle(
```
Replace the TextStyle container with the usual variable boldInter14 or mediumInter14

NN171 replied:
Has been changed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+            },
+            singleLine = true,
+            visualTransformation = if (passwordVisible) VisualTransformation.None
+            else PasswordVisualTransformation(),
```
Add tabs to improve readability

NN171 replied:
Tabs has been added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+            trailingIcon = {
+                IconButton(onClick = {
+                    passwordVisible = !passwordVisible
+                }) {
```
Move ") {" to the next line

NN171 replied:
Has been reformatted

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+                    )
+                }
+            },
+            keyboardOptions = KeyboardOptions(
```
You have only one action with modifier, so you should write on one line.
Example:
```Kotlin
param = function(anotherParam)
```

NN171 replied:
Has been changed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+    val isFieldEmpty by remember {
+        derivedStateOf { text.isEmpty() }
+    }
+    val dimensions = LocalDimensions.current
```
Missed an empty line after this line

NN171 replied:
An empty line has been added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+            },
+            label = {
+                Text(
+                    type,
```
What it this variable? I have already written about this in the PasswordField.kt

NN171 replied:
The param was added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+            label = {
+                Text(
+                    type,
+                    style = if (!isFocused || !isFieldEmpty) TextStyle(
```
Replace the TextStyle container with the usual variable boldInter12 or mediumInter12

NN171 replied:
The method TextStyle has been replaced by the values

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+                        fontSize = 12.sp,
+                        fontFamily = InterFontFamily
+                    )
+                    else TextStyle(
```
Replace the TextStyle container with the usual variable boldInter14 or mediumInter14

NN171 replied:
The method TextStyle has been replaced by the values

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,88 @@
+                focusedIndicatorColor = Color.Transparent
+            ),
+
+            shape = RoundedCornerShape(15.dp),
```
Replace dp value from Dimensions.kt (add dimension if necessary)

NN171 replied:
Has been replaced

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,112 @@
+                unfocusedTrailingIconColor = MaterialTheme.colorScheme.onSecondary
+            ),
+
+            shape = RoundedCornerShape(15.dp)
```
Replace dp value from Dimensions.kt (add dimension if necessary)

NN171 replied:
Has been replaced

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+fun StudentIdTextField(type: String) {
+    var text by remember { mutableStateOf(TextFieldValue()) }
+    var isFocused by remember { mutableStateOf(false) }
+    val textLength = 6
```
Changed to 8 digits

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+    var text by remember { mutableStateOf(TextFieldValue()) }
+    var isFocused by remember { mutableStateOf(false) }
+    val textLength = 6
+    TextField(
```
Empty line was added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+                shape = RoundedCornerShape(15.dp)),
+        value = text,
+        onValueChange = {
+                        if (it.text.length <= textLength) text = it
```
Fixed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+                        if (it.text.length <= textLength) text = it
+                        },
+        textStyle = boldInter14,
+        label = { Text(type,
```
Reformed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+                        },
+        textStyle = boldInter14,
+        label = { Text(type,
+            style = if (!isFocused) TextStyle(
```
There's only one method, so I can use if-else without curly brackets

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+                fontSize = 14.sp,
+                fontFamily = InterFontFamily
+            )
+            ) },
```
Reformed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,73 @@
+                        },
+        textStyle = boldInter14,
+        label = { Text(type,
+            style = if (!isFocused) TextStyle(
```
The TextStyle was replaced by the font type

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/SignUpRedirect.kt:
```diff
@@ -0,0 +1,26 @@
+        color = Color.White,
+        textDecoration = TextDecoration.Underline
+    )
+}
```
The method was deleted

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,93 @@
+            .fillMaxWidth()
+            .onFocusChanged { isFocused = !isFocused }
+            .border(
+                BorderStroke(width = 3.dp, color = Color.Transparent),
```
BorderStroke() was deleted, width and color are now using as parameters

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,93 @@
+        value = password,
+        onValueChange = {password = it},
+        label = { Text(type,
+            style = if (!isFocused) TextStyle(
```
The boldInter10 was replaced by mediumInter10

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,93 @@
+                fontSize = 14.sp,
+                fontFamily = InterFontFamily
+            )
+        ) },
```
Code has been reformatted

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,93 @@
+        ),
+        shape = RoundedCornerShape(15.dp)
+    )
+}
```
The empty line has been added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,93 @@
+        ) },
+        singleLine = true,
+        visualTransformation = if (passwordVisible) VisualTransformation.None
+        else PasswordVisualTransformation(),
```
Has been reformed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/AccountMissing.kt:
Has been changed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/AccountMissing.kt:
I did it for better code readability

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/BoxLabel.kt:
Has been transferred

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/BoxLabel.kt:
Для лучшей читаемости

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/fields/SignUpRedirect.kt:
So, if this file need to be transferred to the navigation package, the button must be transferred to this package too

NN171 replied:
I transferred this file to the components

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,40 @@
+            )
+        }
+    }
+}
```
Has been added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,40 @@
+            .height(50.dp)
+            .width(150.dp)
+            .border(
+                BorderStroke(2.dp, Color.White),
```
It has been replaced

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/SignInButton.kt:
```diff
@@ -0,0 +1,40 @@
+            containerColor = LightBlue
+        )) {
+        Box(contentAlignment = Alignment.Center) {
+            Text(text = "Войти",
```
It was changed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/auth/Login.kt:
```diff
@@ -0,0 +1,86 @@
+            }
+        }
+    }
+}
```
Has been added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/auth/Login.kt:
```diff
@@ -0,0 +1,86 @@
+            .fillMaxSize()
+            .background(Color.White),
+    ) {
+        Column(horizontalAlignment = Alignment.CenterHorizontally,
```
Has been changed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -28,51 +31,56 @@ import com.immortalidiot.rutlead.fields.BoxLabel
+                    .clip(RoundedCornerShape(dimensions.shapeRound))
+                    .background(MaterialTheme.colorScheme.primary)
+                    .border(
+                        BorderStroke(2.dp, MaterialTheme.colorScheme.outline),
```
Has been replaced

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Theme.kt:
```diff
@@ -15,28 +13,36 @@ import androidx.compose.ui.platform.LocalContext
-    */
 )
 
+
```
Has been removed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Theme.kt:
```diff
@@ -15,28 +13,36 @@ import androidx.compose.ui.platform.LocalContext
 )
 
+
+
```
Has been removed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Color.kt:
```diff
@@ -16,6 +16,47 @@ val classicRed = Color.Red
 
 val primaryGrayLightTheme = Color(0xFF878787)
 val primaryGrayDarkTheme = Color(0xFFC8C8C8)
+val DarkWhite = Color(0xFFC7D2E5)
```
Has been changed to lower case

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Color.kt:
Has been done

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+@Preview (showBackground = true)
+@Composable
+fun LoginDesign() {
+    val dimensions = LocalDimensions.current
```
Has been added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.surface),
+    ) {
+        Column(horizontalAlignment = Alignment.CenterHorizontally,
```
Has been reformatted

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                    .clip(RoundedCornerShape(dimensions.shapeXXLRound))
+                    .background(MaterialTheme.colorScheme.primary)
+                    .border(
+                        BorderStroke(dimensions.borderXSWidth, MaterialTheme.colorScheme.outline),
```
Has been replaced

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    modifier = Modifier
+                        .fillMaxWidth(0.85f)
```
Has been changed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                    SignInButton()
+                    Spacer(modifier = Modifier.height(dimensions.verticalXLarge))
+                    Row(modifier = Modifier
+                        .fillMaxWidth(),
```
Has been changed

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+            }
+        }
+    }
+}
```
Was added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Color.kt:
```diff
@@ -16,6 +16,47 @@ val classicRed = Color.Red
+        textSelection = primaryDarkBlue,
+        handle = DarkWhite
+    )
+}
```
Was added

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                    Spacer(modifier = Modifier.height(dimensions.verticalXLarge))
+                    Row(modifier = Modifier
+                        .fillMaxWidth(),
+                        horizontalArrangement = Arrangement.Center) {
```
Has been reformatted

NN171 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,93 @@
+            Image(
+                painter = painterResource(
+                    id = if (isSystemInDarkTheme()) R.drawable.ic_app_dark_logo
+                    else R.drawable.ic_app_light_logo
```
Was added

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/components/login/BoxLabel.kt:
Delete unused imports

NN171 replied:
Has been removed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/components/login/SignUpRedirect.kt:
Delete unused imports

NN171 replied:
Has been removed

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
Why is the signleLine parameter specified in the StudentIDField, but not in the PasswordField? Explain your decision to me

NN171 replied:
SingleLine parameter was set as "true" in both files early.

ImmortalIdiot reviewed (changes requested):
You did a very cool job. I even have a lot to learn. But there are a couple of nuances that I wrote in this review.

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/MainActivity.kt:
```diff
@@ -3,44 +3,16 @@ package com.immortalidiot.rutlead
-    RUTLeadTheme {
-        Greeting("Android")
-    }
-}
```
Add new line

NN171 replied:
Already existing

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/login/SignInButton.kt:
```diff
@@ -0,0 +1,43 @@
+import com.immortalidiot.rutlead.ui.theme.boldLato20
+
+@Composable
+fun SignInButton() {
+    Button(
+        onClick = { /*TODO*/ },
+        modifier = Modifier
```
```suggestion
fun SignInButton(
    onButtonClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Button(
        onClick = onButtonClick,
        modifier = modifier
```

LittlePony00 replied:
Сейчас вы не можете вызвать внутреннюю функцию onClick в Button из вне, с помощью callback onButtonClick, который вы будете передавать в вызываемую функцию, вы сможете :)

Насчёт modifier в параметрах функции SignInButton: благодарю этой передачи, вы сможете в любой момент изменить параметры этой кнопки, её ширину, высоту и т.д. Это считается хорошей практикой, когда функция является универсальной и ее можно изменить в любой момент

ImmortalIdiot replied:
Thank you for advice

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/login/SignInButton.kt:
```diff
@@ -0,0 +1,43 @@
+    ) {
+        Box(contentAlignment = Alignment.Center) {
+            Text(
+                text = "Войти",
```
Тоже самое, добавьте в параметры функции SignInButton какой-нибудь buttonText, таким образом можно будет менять его в момент вызова где угодно, также можно изменить название функции SignInButton на какую-нибудь PrimaryButton, т.к теперь эта функция более универсальна, и вы можете ее кастомизировать где угодно

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/components/login/SignUpRedirect.kt:
```diff
@@ -0,0 +1,24 @@
+    Text(
+        modifier = Modifier
+            .padding(start = LocalDimensions.current.verticalSmallPadding)
+            .clickable { /*TODO*/ },
```
```suggestion
            .clickable { onTextClick() },
```

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/components/login/SignUpRedirect.kt:
```diff
@@ -0,0 +1,24 @@
+@Composable
+fun RedirectText(text: String) {
+    Text(
+        modifier = Modifier
```
```suggestion
        modifier = modifier
```

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,100 @@
+        handleColor = MaterialTheme.colorScheme.scrim,
+        backgroundColor = MaterialTheme.colorScheme.onBackground
+    )
+    var password by rememberSaveable { mutableStateOf("") }
+    var passwordVisible by remember { mutableStateOf(false) }
+    var isFocused by remember { mutableStateOf(false) }
+    val dimensions = LocalDimensions.current
+    val isFieldEmpty by remember {
+        derivedStateOf { password.isEmpty() }
+    }
```
```suggestion
    var password by rememberSaveable { mutableStateOf("") }
    var passwordVisible by remember { mutableStateOf(false) }
    var isFocused by remember { mutableStateOf(false) }
    val isFieldEmpty by remember { derivedStateOf { password.isEmpty() } }
```

LittlePony00 replied:
Нету единого стиля: by remember {} где то на одной строке, где то на двух
+ лучше разделять переменные, которые делают разные задачи: 
dimensons для отступов, что она делает среди remember

ImmortalIdiot replied:
Ok, thank you

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,100 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PasswordField(type: String) {
+    val customCursorHandleColor = TextSelectionColors(
```
```suggestion
fun PasswordField(
    type: String
    modifier: Modifier = Modifier
) {
    val customCursorHandleColor = TextSelectionColors(
```

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,100 @@
+
+    CompositionLocalProvider(LocalTextSelectionColors provides customCursorHandleColor) {
+        TextField(
+            modifier = Modifier
```
```suggestion
            modifier = modifier
```

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,100 @@
+        handleColor = MaterialTheme.colorScheme.scrim,
+        backgroundColor = MaterialTheme.colorScheme.onBackground
+    )
+    var password by rememberSaveable { mutableStateOf("") }
```
Не стоит использовать remember для всего, здесь, например, у вас изменяется текст и вы это запоминаете, но ведь у вас это запоминание происходит во viewModel в функции changeLogin & changePassword. Так есть ли смысл запоминать это и там, и тут, не вызовет ли это ошибок?

ImmortalIdiot replied:
There is no viewMovel for LoginScreen yet

LittlePony00 replied:
Это не значит, что нужно делать неправильно, а вдруг вы забудете?)

Если хочется проверить функционал, то есть @Preview

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,100 @@
+                    )
+                }
+            },
+            singleLine = true,
```
А если вам нужно будет где нибудь использовать этот текст, где одной линии будет недостаточно? Не лучше ли сделать это изменяемым параметром?

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,80 @@
+        handleColor = MaterialTheme.colorScheme.scrim,
+        backgroundColor = MaterialTheme.colorScheme.onBackground
+    )
+    var text by remember { mutableStateOf("") }
```
Тоже самое, что и в PasswordTextField, есть ли смысл запоминать ввод здесь, если он запоминается во viewModel?

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/ui/theme/Theme.kt:
```diff
@@ -15,28 +13,35 @@ import androidx.compose.ui.platform.LocalContext
-    */
 )
 
+
```
Extra empty line

NN171 replied:
Has been removed

ImmortalIdiot reviewed (changes requested):
Actually, Nikita suggested me to make the universal field and just pass the necessary parameters (StudentIDField and Password field replace with a universal field, and when we use this composable function, we will pass the necessary parameters. The same can do with the button.

LittlePony00 commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,4 @@
 <resources>
     <string name="app_name">RUTLead</string>
+    <string name="account_missing">Нет аккаунта? </string>
 </resources>
```
Добавьте пустую строку

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,103 @@
+    var passwordVisible by remember { mutableStateOf(false) }
+    var isFocused by remember { mutableStateOf(false) }
+    val isFieldEmpty by remember { derivedStateOf { password.isEmpty() } }
+    val icon = if (passwordVisible) painterResource(id = R.drawable.password_visibility_on)
```
К сожалению, painterResource не использует remember, что делает вашу функцию restartable, из-за этого при каждом изменение чего-либо в этой функции или на экране, где вы ее будете вызывать, эта иконка будет рекомпозироваться. Альтернативой является ImageVector.vectorResource

Также можно почитать статью на эту тему: https://engineering.teknasyon.com/reduce-recomposition-for-images-icons-in-jetpack-compose-8d2dd3bfa933

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,97 @@
+            verticalArrangement = Arrangement.Center
+        ) {
+            Image(
+                painter = painterResource(
```
Тоже самое, не следует использовать painterResource, лучше заменить его на ImageVector.vectorResource
https://engineering.teknasyon.com/reduce-recomposition-for-images-icons-in-jetpack-compose-8d2dd3bfa933

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,80 @@
+    )
+    var text by remember { mutableStateOf("") }
+    var isFocused by remember { mutableStateOf(false) }
+    val textLength = 8
```
Может лучше вынести в параметр функции? Чтобы сделать эту функцию универсальной и заменить прежнее ее имя на PrimaryTextField, например

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/StudentIdField.kt:
```diff
@@ -0,0 +1,80 @@
+                            else mediumInter14
+                )
+            },
+            singleLine = true,
```
Может лучше не всегда передавать true, а вынести в параметры функции и заменить ее имя на более универсальное?

ImmortalIdiot commented:
I refactored screen, added basic field and button, added view model and custom auth exception functions. Remains to make a snackbar and send the error text to it, add a transition to the sign up screen, open a dialog to reset the password and add sending a request to the server for user authorization.

ImmortalIdiot reviewed (commented):
I refactored screen, added the basic field and the basic button, added the view model and the custom auth exception functions. Remains to make a snackbar and send the error text to it, add a transition to the sign up screen, open a dialog to reset the password and add sending a request to the server for user authorization.

ImmortalIdiot reviewed (commented):
Irrelevant data

ImmortalIdiot reviewed (commented):
I don't know why github is requesting my review again...

ImmortalIdiot reviewed (commented):
All outdated changed mark as resolved

ImmortalIdiot reviewed (commented):
I refactored screen, added the basic field and the basic button, added the view model and the custom auth exception functions. Remains to make a snackbar and send the error text to it (I will add this in this pr), add a transition to the sign up screen, open a dialog to reset the password and add sending a request to the server for user authorization.

ImmortalIdiot commented:
I figured out why github displays me as requesting changes. Instead of clicking on "Approve", I clicked on "Comment". I'm dumb.

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/buttons/login/PrimaryButton.kt:
```diff
@@ -0,0 +1,51 @@
+    modifier: Modifier,
+    palette: ThemeColors,
+    text: String,
+    onButtonClick: () -> Unit,
```
```suggestion
    onButtonClick: () -> Unit
```

ImmortalIdiot replied:
Comma has been removed

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/fields/PasswordField.kt:
```diff
@@ -0,0 +1,143 @@
+        backgroundColor = palette.handleBackground
+    )
+
+    var isPasswordVisible by remember { mutableStateOf(true) }
```
во viewmodel вынести

ImmortalIdiot replied:
К полю разве нужна вью модель?

ImmortalIdiot replied:
Has been done, thank you so much

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/providers/LocalSnackbarHostState.kt:
```diff
@@ -0,0 +1,18 @@
+        withDismissAction = true,
+        duration = SnackbarDuration.Long
+    )
+}
```
```suggestion
}

```

ImmortalIdiot replied:
Has been added, thx

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,209 @@
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusManager = LocalFocusManager.current
+
+    val studentIDErrorMessage = (state as?
+            LoginScreenViewModel.State.ValidationError)?.studentIDError.toString()
```
```suggestion
    val studentIDErrorMessage = (state as? LoginScreenViewModel.State.ValidationError)?.studentIDError.toString()
```

ImmortalIdiot replied:
Reformat code doesn't update tabs

LittlePony00 commented on app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:
```diff
@@ -0,0 +1,209 @@
+
+    val studentIDErrorMessage = (state as?
+            LoginScreenViewModel.State.ValidationError)?.studentIDError.toString()
+    val passwordErrorMessage = (state as?
+            LoginScreenViewModel.State.ValidationError)?.passwordError.toString()
```
```suggestion
    val passwordErrorMessage = (state as? LoginScreenViewModel.State.ValidationError)?.passwordError.toString()
```

ImmortalIdiot replied:
The same story

LittlePony00 commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,5 @@
     <string name="app_name">RUTLead</string>
+    <string name="account_missing">Нет аккаунта? </string>
+
 </resources>
```
```suggestion
</resources>

```

ImmortalIdiot replied:
Has been added, thx

ImmortalIdiot commented on app/src/main/java/com/immortalidiot/rutlead/MainActivity.kt:
```diff
@@ -3,44 +3,48 @@ package com.immortalidiot.rutlead
+                val keyboardController = LocalSoftwareKeyboardController.current
+
+                val palette = if (isSystemInDarkTheme()) ThemeColors.Dark
+                else ThemeColors.Light
```
Has been reformatted, thx
